egs/thchs30/s5/run.sh

```diff
@@ -0,0 +1,113 @@
+#!/bin/bash
+
+. ./cmd.sh ## You'll want to change cmd.sh to something that will work on your system.
+           ## This relates to the queue.
+. ./path.sh
+
+H=`pwd`  #exp home
+n=8      #parallel jobs
+
+thchs=/nfs/public/materials/data/thchs30-openslr
+#corpus and trans directory
+#you can obtain the database by uncommting the following lines
+#( cd $thchs
+#     echo "downloading THCHS30 at $PWD ..."
+#     wget http://www.openslr.org/resources/18/data_thchs30.tgz
+#     wget http://www.openslr.org/resources/18/resource.tgz
+#     tar xvf data_thchs30.tgz && tar xvf resource.tgz 
+#)
+
```

Comment:
Yes, definitely.  Please see how it's done in the Librispeech recipe for an
example.  You don't have to be as careful e.g. with the md5 checksums, but
at least check that it hasn't been downloaded already.
Dan


On Mon, Feb 1, 2016 at 11:45 AM, jtrmal <notifications@github.com> wrote:

> In egs/thchs30/s5/run.sh
> <https://github.com/kaldi-asr/kaldi/pull/464#discussion_r51443469>:
>
> > +           ## This relates to the queue.
> > +. ./path.sh
> > +
> > +H=`pwd`  #exp home
> > +n=8      #parallel jobs
> > +
> > +thchs=/nfs/public/materials/data/thchs30-openslr
> > +#corpus and trans directory
> > +#you can obtain the database by uncommting the following lines
> > +#( cd $thchs
> > +#     echo "downloading THCHS30 at $PWD ..."
> > +#     wget http://www.openslr.org/resources/18/data_thchs30.tgz
> > +#     wget http://www.openslr.org/resources/18/resource.tgz
> > +#     tar xvf data_thchs30.tgz && tar xvf resource.tgz
> > +#)
> > +
>
> The previous commented lines are not completely correct (after
> uncommenting) -- the $thchs variable will be either unset or set to come
> directory that won't probably be available anywhere else than your system.
> I'd say it would make more sense to write:
>
> thchs=/nfs/public/materials/data/thchs30-openslr
> if [ ! -d $thchs ] ; then
> thchs=./thchs30-openslr
> [ -d $thchs ] || mkdir $thchs
> cd $thchs
> echo "downloading THCHS30 at $PWD ..."
> wget http://www.openslr.org/resources/18/data_thchs30.tgz
> wget http://www.openslr.org/resources/18/resource.tgz
> tar xvf data_thchs30.tgz && tar xvf resource.tgz
> fi
>
> eventually you can add some more checks to avoid the downloading and
> untaring the data each time it's the recipe is run.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/464/files#r51443469>.
>